flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/ResourceManagerITCase.java

```diff
@@ -0,0 +1,199 @@
+
+			// check if we registered the task manager resource
+			resourceManager.tell(new TestingResourceManager.GetRegisteredResources(), me);
+
+			TestingResourceManager.GetRegisteredResourcesReply reply =
+				expectMsgClass(TestingResourceManager.GetRegisteredResourcesReply.class);
+
+			assertEquals(1, reply.resources.size());
+			assertTrue(reply.resources.contains(resourceID));
+
+		}};
+		}};
+	}
+
+	/**
+	 * Tests whether the resource manager gets informed upon TaskManager registration.
+	 */
+	@Test
+	public void testResourceManagerTaskManagerRegistration() {
+
+		new JavaTestKit(system){{
+		new Within(duration("30 seconds")) {
+		@Override
+		protected void run() {
+
+			ActorGateway jobManager = TestingUtils.createJobManager(system, config);
+			ActorGateway me =
+				TestingUtils.createForwardingActor(system, getTestActor(), Option.<String>empty());
+
+			// start the resource manager
+			ActorGateway resourceManager =
+				TestingUtils.createResourceManager(system, jobManager.actor(), config);
+
+			// notify about a resource manager registration at the job manager
+			resourceManager.tell(new TestingResourceManager.NotifyWhenResourceManagerConnected(), me);
+
+			// Wait for resource manager
+			expectMsgEquals(Messages.getAcknowledge());
+
+			// start task manager and wait for registration
+			ActorGateway taskManager =
+				TestingUtils.createTaskManager(system, jobManager.actor(), config, true, true);
+
+			// check if we registered the task manager resource
+			resourceManager.tell(new TestingResourceManager.GetRegisteredResources(), me);
+
+			TestingResourceManager.GetRegisteredResourcesReply reply =
+				expectMsgClass(TestingResourceManager.GetRegisteredResourcesReply.class);
+
+			assertEquals(1, reply.resources.size());
+
+		}};
+		}};
+	}
+
+
+	/**
+	 * Tests cluster shutdown procedure of RM
+	 */
+	@Test
+	public void testClusterShutdown() {
+		new JavaTestKit(system){{
+			new Within(duration("30 seconds")) {
+				@Override
+				protected void run() {
+
+					ActorGateway me =
+						TestingUtils.createForwardingActor(system, getTestActor(), Option.<String>empty());
+
+					ActorGateway jobManager = TestingUtils.createJobManager(system, config);
+
+					ActorGateway resourceManager =
+						TestingUtils.createResourceManager(system, jobManager.actor(), config);
+
+					// notify about a resource manager registration at the job manager
+					resourceManager.tell(new TestingResourceManager.NotifyWhenResourceManagerConnected(), me);
+
+					// Wait for resource manager
+					expectMsgEquals(Messages.getAcknowledge());
+
+					jobManager.tell(new StopCluster(ApplicationStatus.SUCCEEDED, "Shutting down."), me);
+
+					expectMsgClass(StopClusterSuccessful.class);
+
+					boolean isTerminated = false;
+					for (int i=0; i < 10 && !isTerminated; i++) {
+						isTerminated = system.isTerminated();
+						try {
+							Thread.sleep(1000);
+						} catch (InterruptedException e) {
+							// try again
+						}
+					}
```

Comment:
True, I mistook that for a `BeforeClass` and `AfterClass`. Wouldn't it then make sense to put all tests which can work using the same `ActorSystem` in one class using `AfterClass`/`BeforeClass` and leave the tests which require a fresh `ActorSystem` here? Bringing up an `ActorSystem` is quite costly. Especially wrt to long test runtimes this could be helpful.